scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -583,6 +583,7 @@ class ScioContext private[scio] (
 
   private[scio] def execute(): ScioExecutionContext = {
     val sc = this
+    if (isTest) TestDataManager.overrideTransforms(testId.get, pipeline)
```

Comment:
:art: here I would move this to https://github.com/deacondesperado/scio/tree/dd1b576a41014f062546a51af63f01024b0fa4e5/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

Reply:
@regadas The overrides need to be applied after the entire pipeline is constructed, so I don't think applying the overrides here would work.